metals/src/main/scala/scala/meta/internal/metals/WorkspaceSearchVisitor.scala

```diff
@@ -132,7 +132,7 @@ class WorkspaceSearchVisitor(
       isVisited += defn.path
       val input = defn.path.toInput
       SemanticdbDefinition.foreach(input, defn.dialect) { semanticDefn =>
-        if (query.matches(semanticDefn.info)) {
+        if (query.matches(semanticDefn.info, input.toLanguage.isScala)) {
```

Comment:
This shouldn't be needed, we can also have Java files in the workspace

Reply:
On the contrary. I need to add it in other places too. Java mtags produce also methods and they get filtered by `isRelevantKind`, which definition (for Scala) has changed.

Reply:
Sorry for coming back to it now. So we will not want to index java methods? Or was it because we didn't want to include the ones from classes. With current approach we can skip this check, right?

Reply:
Thanks for bringing this up, you're quite right.